client.go

```diff
@@ -192,9 +179,13 @@ func (c *Client) request(stream string) (*http.Response, error) {
 	return c.Connection.Do(req)
 }
 
-func (c *Client) processEvent(msg []byte) *Event {
+func (c *Client) processEvent(msg []byte) (event *Event, err error) {
```

Comment:
1. Made `processEvent` return a proper error, instead of a nil `*Event`.
2. Moved the check for a valid incoming `msg` into this function, so we didn't have to check it before passing it in. Seemed like the right thing.
3. Remove the `return nil` in `default` case of the for loop. That was wrong. We should ignore the line and keep processing when we get a line that doesn't match.